PortalSwift/Sources/PortalSwift/Utils/HttpRequest.swift

```diff
@@ -116,7 +117,14 @@ public class HttpRequest<T: Codable, BodyType> {
               } else {
                 typedData = try JSONDecoder().decode(T.self, from: data!)
               }
-              return completion(Result(error: GatewayError.gatewayError(response: (typedData as! ETHGatewayResponse).error!, status: String(httpResponse!.statusCode))))
+              return completion(
+                Result(
+                  error: GatewayError.gatewayError(
+                    response: typedData as! PortalProviderRpcResponseError,
```

Comment:
should we force cast this typedData? If it is not the correct type will this crash the app?

Reply:
ah i see we are doing force casting all of this class 😭 so you are really just changing the type from `ETHGatewayResponse` to `PortalProviderRpcResponseError`

Reply:
Could we do somthing like you have below here? 
```suggestion
                                        response: (typedData as? PortalProviderRpcResponseError) ?? PortalProviderRpcResponseError(code: 32602, message: "Unknown Error"),
```

Reply:
If we don't get a valid response, this will throw before `typedData` is defined, so there's no need to add an additional check.

Reply:
If this code `typedData = try JSONDecoder().decode(T.self, from: data!)` decodes the response and it isn't a `PortalProviderRpcResponseError` then wouldn't we crash?

Reply:
Since it's a `try`, it will just throw the error, which will bubble up.

In either case, this code:
1. Hasn't changed at all
2. Is no longer used by the SDK. It's only still in the repo because `Cocoapods Example` uses it still.